.github/workflows/CI.yml

```
@@ -182,3 +182,66 @@ jobs:
         with:
           key: ${{ steps.build-cache.outputs.cache-primary-key }}
           path: ~/.cache
+  release:
+    name: "Publish Pre-Release"
+    runs-on: ubuntu-20.04-16core
+    needs: [test]
```

As documented by https://docs.github.com/en/rest/actions/artifacts?apiVersion=2022-11-28#list-workflow-run-artifacts
```suggestion
    needs: [test]
    permissions:
      actions: read
```